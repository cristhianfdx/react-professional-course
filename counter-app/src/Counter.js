import React, { useState } from 'react';
import PropTypes from 'prop-types';

const Counter = ({ value = 10 }) => {
  const [counter, setCounter] = useState(value);

  const handleClick = () => {
    setCounter(counter + 1);
  };

  const handleReset = () => setCounter(value);

  const handleSubtract = () => setCounter(counter - 1);

  return (
    <div className='text-center'>
      <h1>Counter</h1>
      <h2>{counter}</h2>
      <button className='btn btn-primary mr-2' onClick={handleClick}>
        +1
      </button>
      <button className='btn btn-secondary mr-2' onClick={handleReset}>
        Reset
      </button>
      <button className='btn btn-danger' onClick={handleSubtract}>
        -1
      </button>
    </div>
  );
};

Counter.propTypes = {
  value: PropTypes.number,
};

export default Counter;
