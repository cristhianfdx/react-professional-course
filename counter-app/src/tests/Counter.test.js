import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import React from 'react';

import Counter from '../Counter';

describe('Counter Component Tests', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Counter />);
  });

  test('should to show component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should show the default value of 100', () => {
    const value = 100;
    const wrapper = shallow(<Counter value={value} />);

    const counterValue = wrapper.find('h2').text().trim();

    expect(counterValue).toBe(value.toString());
  });

  test('should increment the counter', () => {
    wrapper.find('button').at(0).simulate('click');
    const counterValue = wrapper.find('h2').text().trim();
    expect(counterValue).toBe('11');
  });

  test('should decrease the counter', () => {
    wrapper.find('button').at(2).simulate('click');
    const counterValue = wrapper.find('h2').text().trim();
    expect(counterValue).toBe('9');
  });

  test('should reset the counter', () => {
    const wrapper = shallow(<Counter value={100} />);
    wrapper.find('button').at(1).simulate('click');
    const counterValue = wrapper.find('h2').text().trim();
    expect(counterValue).toBe('100');
  });
});
