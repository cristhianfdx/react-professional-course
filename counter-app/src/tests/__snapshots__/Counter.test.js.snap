// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Tests should to show component correctly 1`] = `
<div
  className="text-center"
>
  <h1>
    Counter
  </h1>
  <h2>
    10
  </h2>
  <button
    className="btn btn-primary mr-2"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    className="btn btn-secondary mr-2"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    -1
  </button>
</div>
`;
